tests/func/test_checkout.py

```diff
@@ -475,3 +479,48 @@ def test_checkout_no_checksum(repo_dir, dvc_repo):
     stage = dvc_repo.run(outs=[repo_dir.FOO], no_exec=True, cmd="somecmd")
     dvc_repo.checkout(stage.path, force=True)
     assert not os.path.exists(repo_dir.FOO)
+
+
+def test_should_not_checkout_when_adding_cached_copy(repo_dir, dvc_repo):
+    dvc_repo.cache.local.cache_types = ["copy"]
+
+    dvc_repo.add(repo_dir.FOO)
+    dvc_repo.add(repo_dir.BAR)
+
+    shutil.copy(repo_dir.BAR, repo_dir.FOO)
+
+    remove_spy = spy(dvc.remote.local.RemoteLOCAL.remove)
+    with patch.object(dvc.remote.local.RemoteLOCAL, "remove", remove_spy):
```

Comment:
Why are you testing `remove()` and not `copy()`. It's possible to implement this as "copy over", while not breaking this test.

Reply:
Done, will check copyfile calls